syntax = "proto3";

package user;

// User service definition
service UserService {
  // Get user information
  rpc GetUser (UserRequest) returns (UserResponse);
  
  // Create a new user
  rpc CreateUser (CreateUserRequest) returns (UserResponse);
  
  // List all users
  rpc ListUsers (Empty) returns (UserListResponse);
}

// Empty message for requests with no parameters
message Empty {}

// Request message for getting a user
message UserRequest {
  int32 user_id = 1;
}

// Request message for creating a user
message CreateUserRequest {
  string name = 1;
  string email = 2;
  int32 age = 3;
}

// Response message containing user information
message UserResponse {
  int32 user_id = 1;
  string name = 2;
  string email = 3;
  int32 age = 4;
  string created_at = 5;
}

// Response message containing a list of users
message UserListResponse {
  repeated UserResponse users = 1;
}
